src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -778,13 +778,15 @@ class LicenceService(
     val createdBy = this.communityOffenderManagerRepository.findByUsernameIgnoreCase(username)
       ?: throw IllegalStateException("Cannot find staff with username: $username")
 
-    val licenceCopy = licence.copyLicence(newStatus)
+    val newLicenceVersion =
+      licence.licenceVersion?.let { getNextLicenceVersion(it, newStatus) }
+
+    val licenceCopy = licence.copyLicence(newStatus, newLicenceVersion)
```

Comment:
this method is humongous now!..

Reply:
yes, will have a look to see if there is anything we can extract out